scripts/Muon/GUI/Common/fitting_tab_widget/fitting_tab_presenter.py

```diff
@@ -44,7 +61,82 @@ def handle_use_rebin_changed(self):
         if not self.manual_selection_made:
             self.update_selected_workspace_guess()
         else:
-            self.selected_data = self.context.get_list_of_binned_or_unbinned_workspaces_from_equivalents(self.selected_data)
+            self.selected_data = self.context.get_list_of_binned_or_unbinned_workspaces_from_equivalents(
+                self.selected_data)
+
+    def handle_fit_type_changed(self):
+        fit_type = self.view.fit_type
+
+        if fit_type == 'Single Fit':
+            self.view.workspace_combo_box_label.setText('Select Workspace')
+            if not self.manual_selection_made:
+                guess_selection = self.context.get_names_of_workspaces_to_fit(runs='All',
+                                                                              group_and_pair='All', phasequad=True,
+                                                                              rebin=not self.view.fit_to_raw)
+            else:
+                guess_selection = self.selected_data
+            self.selected_data = guess_selection
+        else:
+            self.view.workspace_combo_box_label.setText('Display parameters for')
+            if not self.manual_selection_made:
+                guess_selection = self.context.get_names_of_workspaces_to_fit(runs='All',
+                                                                              group_and_pair=self.context.gui_context[
+                                                                                  'selected_group_pair'], phasequad=True,
+                                                                              rebin=not self.view.fit_to_raw)
+            else:
+                guess_selection = self.selected_data
+
+            self.selected_data = guess_selection
+
+    def handle_fit_clicked(self):
+        fit_type = self.view.fit_type
+
+        try:
+            if fit_type == 'Single Fit':
```

Comment:
lets make the 'single fit' etc. const strings, probably best to define them in the view and have a way to access the names from here.